### PR TITLE
Application Templates Documentation

### PR DESCRIPTION
**Pull request by awholegunch**
First cut at documenting app template details.

Tries to standardize the descriptions located in the source code vis-a-vis style and punctuation.

Staged at: [ETL Template and Plugins](http://docs-staging.cask.co/cdap/3.1.0-SNAPSHOT-feature-r3.1_Doc_Plugins/en/application-templates/etl/templates/index.html)

Note: adding *?hidenav* to the end of page URLs will hide all the navigation, such as
- [complete](http://docs-staging.cask.co/cdap/3.1.0-SNAPSHOT-feature-r3.1_Doc_Plugins/en/application-templates/etl/templates/sources/batch/database.html)
- [hidden](http://docs-staging.cask.co/cdap/3.1.0-SNAPSHOT-feature-r3.1_Doc_Plugins/en/application-templates/etl/templates/sources/batch/database.html?hidenav)

This feature of the documentation will be documented on the wiki...

Future PRs will add to this material, but this will be the initial scaffolding for both the documentation and the UI.

### CODE REVIEW

File: cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/batch/sink/DBSink.java [hunk 1]
```diff
@@ -173,7 +173,7 @@ private void ensureJDBCDriverIsAvailable() throws Exception {
       DriverManager.getDriver(dbSinkConfig.connectionString);
     } catch (SQLException e) {
       // Driver not found. We will try to register it with the DriverManager.
-      LOG.debug("Plugin Type: {} and Plugin Name: {}; Driver Class: {} not found. Registering JDBC driver via shim {} ",
+      LOG.debug("Plugin Type: {} and Plugin Name: {}; Driver Class: {} not found; Registering JDBC driver via shim {} ",
```
Comment (anwar6953) [inline cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/batch/sink/DBSink.java]: should Registering be lowercase 'r' then?

File: cdap-docs/_common/_themes/cdap/static/cdap.css_t [hunk 2]
```diff
@@ -454,3 +454,13 @@ div.table-block-example.container table.triple-table.docutils tr:first-of-type {
...
+
+/* -- Styles to hide navigation --- */
+
+div.documentwrapper-hidden {
+    margin-top: 0px;
+}
+
+div.bodywrapper-hidden {
+    margin: 0 0 0 0;
+}
```
Comment (anwar6953) [inline cdap-docs/_common/_themes/cdap/static/cdap.css_t]: add newline

File: cdap-docs/_common/_themes/cdap/static/doctools.js [hunk 3]
```diff
@@ -229,7 +230,22 @@ var Documentation = {
...
-  }
+  },
+  
+  /**
+   * Hide page navigation if requested
+   */
+  hidePageNavigation : function() {
+    var params = $.getQueryParameters();
+    if (params.hidenav) {
+      console.log('hiding navigation');
```
Comment (anwar6953) [inline cdap-docs/_common/_themes/cdap/static/doctools.js]: remove console.log?

**Comment by anwar6953** [comment]
Are the rst files in real-time/ and batch/ directories auto generated, based upon the relevant sinks/sources?

**Comment by awholegunch** [comment]
No, they are hand-edited. It's something we should consider for another release, but I think we need to see what information ends up in the rst files, how it works in both the documentation and in the CDAP ui, and then at that point figure out where to store the information and how to generate it. But it certainly makes sense to have it in only one location, and source it from there. I certainly don't want to have to maintain it two places!

File: cdap-docs/_common/_themes/cdap/static/cdap.css_t (hunk 2 above)
Comment (awholegunch) [inline cdap-docs/_common/_themes/cdap/static/cdap.css_t]: fixed.

File: cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/batch/sink/DBSink.java (hunk 1 above)
Comment (awholegunch) [inline cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/batch/sink/DBSink.java]: fixed.

File: cdap-docs/_common/_themes/cdap/static/doctools.js (hunk 3 above)
Comment (awholegunch) [inline cdap-docs/_common/_themes/cdap/static/doctools.js]: fixed.

**Comment by awholegunch** [comment]
Created https://issues.cask.co/browse/CDAP-3080 to track this.

**Comment by albertshau** [comment]
:+1:

**Pull request merged**